sympy/sets/fancysets.py

```diff
@@ -10,6 +10,7 @@
 from sympy.logic.boolalg import And
 from sympy.sets.sets import Set, Interval, Union, FiniteSet
 from sympy.utilities.misc import filldedent
+from sympy.core.numbers import Float
```

Comment:
I would import `Expr` instead an check that the elements of `Range` are instances of `Expr`.

Reply:
For what purpose?

And you know, I *really* don't like the idea of making `Range` hold things like `Range(1 + sqrt(2), sqrt(117), sqrt(3))`. And even if it does, the end value should be made canonical so that becomes `Range(1 + sqrt(2), sqrt(2) + 7, sqrt(3))`. Would it be better to allow an offset and scale that default to 0 and 1 and only allow start, step and step to be integers? So the previous would be `Range(0, 6, 1, offset=sqrt(2), scale=sqrt(3)) -> start + offset + i*scale`? But do we really need this or is it sufficient for the stats needs just to allow a Symbol that gets replaced with an Integer?

Reply:
I think allowing only `Integers` and `Symbols` is good. We can leave `pi`, `sqrt(2)`, `sqrt(117)` can be done in another PR. I will raise a separate issue for that. What do you say?